comp/dogstatsd/replay/mock/mock.go

```diff
@@ -3,31 +3,47 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2016-2021 Datadog, Inc.
 
-package replay
+//go:build test
+
+//nolint:revive // TODO(AML) Fix revive linter
+package mock
 
 import (
 	"context"
 	"sync"
 	"time"
 
-	"go.uber.org/fx"
-
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	compdef "github.com/DataDog/datadog-agent/comp/def"
 	"github.com/DataDog/datadog-agent/comp/dogstatsd/packets"
+	replaydef "github.com/DataDog/datadog-agent/comp/dogstatsd/replay/def"
 )
 
-type mockTrafficCapture struct {
-	isRunning bool
-	sync.RWMutex
+// Mock implements mock-specific methods.
+type Mock interface {
+	replaydef.Component
 }
 
-func newMockTrafficCapture(deps dependencies) Component {
+//nolint:revive // TODO(AML) Fix revive linter
+type Requires struct {
+	Lc     compdef.Lifecycle
+	Config config.Component
+}
+
+//nolint:revive // TODO(AML) Fix revive linter
+func NewMockTrafficCapture(deps Requires) replaydef.Component {
```

Comment:
we should rename this as just `New` or `NewTrafficCapture` since we're in the mock folder.
Also, a mock should take a `t *testing.T` as parameter, even if we do nothing with this. 
Lastly, we don't need the `Requires`, no ? The mock should help other piece of code test their interaction with the replay.

Reply:
Fixed in [`947e550` (#25512)](https://github.com/DataDog/datadog-agent/pull/25512/commits/947e55021152f85c46dc8d4a7b35f69429715749), I had to keep the `Requires` to work with `ProvideComponentConstructor` in `fx-mock`